src/main/java/com/denizenscript/denizencore/scripts/commands/core/SQLCommand.java

```diff
@@ -323,19 +325,24 @@ else if (action.asString().equalsIgnoreCase("query")) {
                         int count = 0;
                         ListTag rows = new ListTag();
                         ListTag resultList = new ListTag();
+                        ListTag resultMap = new ListTag();
                         while (set.next()) {
                             count++;
                             StringBuilder current = new StringBuilder();
                             ListTag subList = new ListTag();
+                            MapTag subMap = new MapTag();
                             for (int i = 0; i < columns; i++) {
                                 current.append(EscapeTagUtil.escape(set.getString(i + 1))).append("/");
                                 subList.addObject(new ElementTag(set.getString(i + 1)));
+                                subMap.putObject(rsmd.getColumnLabel(i + 1), new ElementTag(set.getString(i + 1)));

```

Comment:
I'd probably put `set.getString(i + 1)` in a variable here as it's being called 3 times, otherwise LGTM.